browser-extension/mv2/src/background/background.js

```diff
@@ -1007,6 +1008,54 @@ BG.Methods.addListenerForExtensionMessages = function () {
   });
 };
 
+BG.Methods.handleClientPortConnections = () => {
+  chrome.runtime.onConnect.addListener((port) => {
+    const tabId = port.sender.tab.id;
+
+    if (port.sender.documentLifecycle === "active") {
+      window.tabService.setData(tabId, BG.TAB_SERVICE_DATA.CLIENT_PORT, port);
+
+      const clientLoadSubscribers = window.tabService.getData(tabId, BG.TAB_SERVICE_DATA.CLIENT_LOAD_SUBSCRIBERS) || [];
+      window.tabService.removeData(tabId, BG.TAB_SERVICE_DATA.CLIENT_LOAD_SUBSCRIBERS);
+      clientLoadSubscribers.forEach((subscriber) => subscriber());
+    }
+
+    port.onDisconnect.addListener(() => {
+      window.tabService.removeData(tabId, BG.TAB_SERVICE_DATA.CLIENT_PORT);
+
+      if (!BG.Methods.isConnectedToClient(tabId)) {
```

Comment:
@lazyvab This part tries to reestablish the connection if it has been disconnected due to navigation? Right?

Reply:
Exactly.